Test/Get-TaskParameter.Tests.ps1

```diff
@@ -0,0 +1,611 @@
+        ThenPipelineSucceeded
+        ThenTaskCalled -WithParameter @{ 'Path' = $context.ConfigurationPath.FullName }
+    }
+}
+
+Describe ('Get-TaskParameter.when parameter is an optional path but it doesn''t exist') {
+    It 'should return a full path' {
+        function global:Task
+        {
+            [Whiskey.Task('Task')]
+            param(
+                [Whiskey.Tasks.ValidatePath()]
+                [string]
+                $Path
+            )
+            $global:taskCalled = $true
+            $global:taskParameters = $PSBoundParameters
+        }
+        Init
+        WhenRunningTask 'Task' -Parameter @{ 'Path' = 'somefile.txt' } -ErrorAction SilentlyContinue
+        ThenTaskNotCalled
+        ThenThrewException 'does\ not\ exist'
+    }
+}
+
+Describe ('Get-TaskParameter.when parameter is a path with wildcards') {
+    It 'should resolve path to actual paths' {
+        function global:Task
+        {
+            [Whiskey.Task('Task')]
+            param(
+                [Whiskey.Tasks.ValidatePath()]
+                [string[]]
+                $Path
+            )
+            $global:taskCalled = $true
+            $global:taskParameters = $PSBoundParameters
+        }
+        Init
+        GivenFile 'abc.yml' 
+        WhenRunningTask 'Task' -Parameter @{ 'Path' = '*.yml' } 
+        ThenPipelineSucceeded
+        ThenTaskCalled
+        $taskParameters['Path'] | Should -HaveCount 2
+        $taskParameters['Path'] | Should -Contain (Join-Path -Path $Context.BuildRoot -ChildPath 'abc.yml')
+        $taskParameters['Path'] | Should -Contain $context.ConfigurationPath.FullName
+    }
+}
+
+Describe ('Get-TaskParameter.when parameter is a path that the user wants resolved with a wildcard but doesn''t exist') {
+    It 'should fail' {
+        function global:Task
+        {
+            [Whiskey.Task('Task')]
+            param(
+                [Whiskey.Tasks.ValidatePath()]
+                [string[]]
+                $Path
+            )
+            $global:taskCalled = $true
+            $global:taskParameters = $PSBoundParameters
+        }
+        Init
+        GivenFile 'abc.yml' 
+        WhenRunningTask 'Task' -Parameter @{ 'Path' = '*.fubar' } -ErrorAction SilentlyContinue
+        ThenThrewException 'does\ not\ exist'
+        ThenTaskNotCalled
+    }
+}
+
+Describe ('Get-TaskParameter.when path parameter wants to be resolved but parameter type isn''t a string array') {
+    It 'should fail' {
+        function global:Task
+        {
+            [Whiskey.Task('Task')]
+            param(
+                [Whiskey.Tasks.ValidatePath()]
+                [string]
+                $Path
+            )
+            $global:taskCalled = $true
+            $global:taskParameters = $PSBoundParameters
+        }
+        Init
+        GivenFile 'abc.yml' 
+        WhenRunningTask 'Task' -Parameter @{ 'Path' = '*.yml' } -ErrorAction SilentlyContinue

```

Comment:
How does this test work? I only see one `*.yml` file being created.

Reply:
The tests create a default whiskey.yml file. Should be clearer now.